ecs-agent/netlib/platform/common_linux.go

```diff
@@ -695,22 +695,31 @@ func (c *common) configureBranchENI(ctx context.Context, netNSPath string, eni *
 		"NetNSPath": netNSPath,
 	})
 
-	var cniNetConf ecscni.PluginConfig
+	c.os.Setenv(IPAMDataPathEnv, filepath.Join(c.stateDBDir, IPAMDataFileName))
+
+	var cniNetConf []ecscni.PluginConfig
 	var err error
 	add := true
 
 	// Generate CNI network configuration based on the ENI's desired state.
 	switch eni.DesiredStatus {
 	case status.NetworkReadyPull:
-		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeVlan)
-	case status.NetworkReady:
-		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeTap)
+		// Setup bridge to connect task network namespace to TMDS running in host's primary netns.
+		if eni.IsPrimary() {
+			cniNetConf = append(cniNetConf, createBridgePluginConfig(netNSPath))
+		}
+		cniNetConf = append(cniNetConf, createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeVlan))
 	case status.NetworkDeleted:
-		cniNetConf = createBranchENIConfig(netNSPath, eni, VPCBranchENIInterfaceTypeTap)
+		// Branch ENIs are cleaned up by the Control Plane. Added log entry to increase debuggability.
+		logger.Debug("Request to clean up branch ENI received", map[string]interface{}{
+			"ENIName":   eni.Name,
+			"NetNSPath": netNSPath,
+		})
```

Comment:
Why don't we need to execute the DELETE CNI action in this code path?

Reply:
The VLAN configuration gets cleaned up when we delete the network namespace.

Reply:
Not quite following this. We will invoke `c.executeCNIPlugin(ctx, add = false, cniNetConf = nil)` in this scenario. How does the VLAN configuration get cleaned up?

Reply:
I see, yes this is a bug I will fix it. 

I assumed the cleanup was unnecessary for two reasons:

1. It looks like the FC cleanup was only removing the TAP interface, however, if take a look at the CNI plugin implementation, we clean up[ both TAP and VLAN link. ](https://github.com/aws/amazon-vpc-cni-plugins/blob/master/plugins/vpc-branch-eni/plugin/commands.go#L238)

2. Once the task stops, I searched for the vlan IF using `ip link list` and I did not see the branch ENI anymore. I was able to run multiple tasks without a vlan id conflict, then I assumed that network namespace deletion was sufficient to clean up the links. I probably just got lucky. 

Next steps:
Modify the Plugin to handle the scenario where we only need to delete branch ENIs